src/Components/App/App.tsx

```diff
@@ -1,22 +1,81 @@
-import './App.css';
-import NavBar from '../NavBar/NavBar';
-import { Route, Routes } from 'react-router-dom';
-import Homepage from '../Homepage/Homepage';
+import "./App.css";
+import React from "react";
+import { useEffect, useState } from "react";
+import { Route, Routes } from "react-router-dom";
+import Homepage from "../Homepage/Homepage";
+import { generateID, cleanUpData, ArtData} from "../../utils";
 import SavedPage from '../SavedPage/SavedPage';
+import NavBar from '../NavBar/NavBar';
+
+function App() {
+  const [pieces, setPieces] = useState<ArtData[]>([]);
+  const [savePieces, setSavePieces] = useState<ArtData[]>([]);
+
+  const getArt = (objectID: number) => {
+    console.log(" obj ID", objectID);
+    return fetch(
+      `https://collectionapi.metmuseum.org/public/collection/v1/objects/${objectID}`
+    ).then((res) => res.json());
+  };
+  useEffect(() => {
+    const callApi = () => {
+      getArt(generateID()).then((data) => {
+        if (data.primaryImage && data.message !== "ObjectID not found") {
+          const cleanData: any = cleanUpData(data);
+          console.log("DATA WITH IMAGE", cleanData);
+          setPieces((prev) => [...prev, cleanData]);
+          return cleanData;
+        } else if (
+          !data.primaryImage ||
+          data.message === "ObjectID not found"
+        ) {
+          callApi();
+        }
+      });
+    };
```

Comment:
Have you considered moving the declaration of the callAPI function outside of the useEffect then simply invoking it within the useEffect?  It's your decision but I'd love to know why or why not you would choose to have it there.

Reply:
We have it in there because we are invoking it within itself to be basically run until we have 3 pieces of “good” data. And we have to invoke the api call function in a use effect right? Not sure what’s best since we also have never tried calling a function within its self